### ISSUE TITLE
Implement the C interface tutorial for Scala

### ISSUE THREAD

**Issue by vjovanov** [oracle/graal#866]
It is important to see how we can interface C from Scala. Will all Java annotations and bytecode intrinsic work out properly in GraalVM. 

We need to implement an equivalent of our [Java-to-native example](https://github.com/oracle/graal/blob/master/substratevm/src/com.oracle.svm.tutorial/src/com/oracle/svm/tutorial/CInterfaceTutorial.java).

**Comment by markehammons** [comment]
I looked at this. Just glancing at it, I'm pretty sure none of the enum examples would work properly, considering how enums are in scala. I usually have to dip into java for enums when using stuff like JNA/JNR.

**Comment by vjovanov** [comment]
Yes, currently `enum`s and annotations on static methods are the biggest issues. I am looking for a way to support them without being overly Scala specific in the image generator.
